theano/sandbox/cuda/opt.py

```diff
@@ -1899,7 +1899,10 @@ def local_gpu_downsample_factor_max(node):
 
         assert node.op.__props__ == ('ds', 'ignore_border', 'st', 'padding',
                                      'mode')
-        if node.op.padding != (0, 0) or node.op.mode != 'max':
+        if (len(node.op.ds) not in (2, 3) or
```

Comment:
This is our old pooling. It support only 2d. So make the condition that node.op.ds == 2.